### PR TITLE
Update Arrow Interface Scan Behavior

### PR DESCRIPTION
**Pull request by sfc-gh-jbahk**
### Description
Please explain the changes you made here.

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: driver_test.go [hunk 1]
```diff
@@ -996,6 +1041,118 @@ func testVariousTypes(t *testing.T, arrow bool) {
 	})
 }
 
+func TestArrowVariousTypes(t *testing.T) {
+	runTests(t, dsn, func(dbt *DBTest) {
+		dbt.mustExec("ALTER SESSION set go_query_result_format = arrow_force")
+		rows := dbt.mustQuery(
+			"SELECT 1.0::NUMBER(30,2) as C1, 2::NUMBER(38,0) AS C2, 't3' AS C3, 4.2::DOUBLE AS C4, 'abcd'::BINARY AS C5, true AS C6")
```
Comment (sfc-gh-stakeda) [inline driver_test.go:1048]: can you split into multiple test cases per data type?

File: driver_test.go [hunk 2]
```diff
@@ -546,6 +547,61 @@ func testInt(t *testing.T, arrow bool) {
 	})
 }
 
+func TestFixedArrow(t *testing.T) {
+	runTests(t, dsn, func(dbt *DBTest) {
+
+		dbt.mustExec("ALTER SESSION set go_query_result_format = arrow_force")
+		//rows := dbt.mustQuery("SELECT 10.0::NUMBER(38,36)")
+		//rows := dbt.mustQuery("SELECT 23456.1::NUMBER(31,20)")
```
Comment (sfc-gh-stakeda) [inline driver_test.go:555]: can you remove these two lines if not required?

File: converter.go [hunk 3]
```diff
@@ -255,35 +256,67 @@ func arrowToValue(destcol *[]snowflakeValue, srcColumnMeta execResponseRowType,
...
-
 		}
 		return err
 	case "BOOLEAN":
 		boolData := array.NewBooleanData(data)
 		for i := range *destcol {
 			if !srcValue.IsNull(i) {
 				(*destcol)[i] = boolData.Value(i)
-			} else {
-				(*destcol)[i] = nil
```
Comment (sfc-gh-stakeda) [inline converter.go:277]: does this mean the destcol is already nil initialized?

File: driver_test.go (hunk 1 above)
Comment (sfc-gh-jbahk) [reply driver_test.go:1048]: This was the Arrow complement to the `TestVariousTypes` JSON test. The test per data type can be found under `TestArrow<data type>`. If you feel this test is unnecessary, then I can remove it.

File: converter.go (hunk 3 above)
Comment (sfc-gh-jbahk) [reply converter.go:277]: Yes

File: driver_test.go (hunk 1 above)
Comment (sfc-gh-stakeda) [reply driver_test.go:1048]: ok, can you add a comment explaining the difference?

Comment (sfc-gh-jbahk) [reply driver_test.go:1048]: Between `TestVariousTypes` and this one?

Comment (sfc-gh-jbahk) [reply driver_test.go:1048]: Done.

File: driver_test.go (hunk 2 above)
Comment (sfc-gh-stakeda) [reply driver_test.go:555]: ok, let's add `23456.1::NUMBER(31,20)` test case to receive decimal128 for now and focus other data type conversions.

**Pull request merged**